src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioOperationServiceImpl.java

```diff
@@ -225,4 +268,46 @@ private Path generateOutputPath(Path basePath,
         return basePath.resolve(filename);
     }
 
+    private List<List<AudioFileInfo>> getSeparatedAudioFileInfo(List<AudioFileInfo> audioFileInfoList, int acceptableAudioGapSecs) {
+
+        if (audioFileInfoList.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        List<List<AudioFileInfo>> audioFileInfoBySessionList = new ArrayList<>();
+        List<AudioFileInfo> sessionAudio = new ArrayList<>();
+        AudioFileInfo previousAudio;
+        AudioFileInfo thisAudio;
+        sessionAudio.add(audioFileInfoList.get(0));
+
+        for (int counter = 1; counter < audioFileInfoList.size(); counter++) {
+            previousAudio = audioFileInfoList.get(counter - 1);
+            thisAudio = audioFileInfoList.get(counter);
+            if (hasGapBetweenAudios(previousAudio, thisAudio, acceptableAudioGapSecs)) {
+                //create new session
+                audioFileInfoBySessionList.add(sessionAudio);
+                sessionAudio = new ArrayList<>();
+            }
+            sessionAudio.add(thisAudio);
+        }
+        audioFileInfoBySessionList.add(sessionAudio);
+        return audioFileInfoBySessionList;
+
+    }
+
+    private boolean hasGapBetweenAudios(AudioFileInfo audioFileInfoFirst, AudioFileInfo audioFileInfoNext,int acceptableAudioGapSecs) {
```

Comment:
Can you check this is giving the result you expect? E.g. if you have a millisecond gap of `1200` `calcGap` will evaulate to `1` because you're doing integer division thereby losing precision on the division operation. The conditional check would then not evaluate to `true`, despite the calcGap actually being larger than the configured allowable of 1 second (1000 ms).

Personally I would stay away from integers/longs here, and instead represent both `acceptableAudioGapSecs` and `calcGap` as java.time.Duration types. This is more expressive, eliminating the need for manual conversion as Durations can be directly compared. Also avoids the need to describe the units in the variable name, minimising chances of misinterpretation.

To solve this problem at it's root, I suggest changing the following value in`application.yaml` from:
```yaml
audio_gap_seconds: 1
```

to 

```yaml
audio_gap_duration: 1ms
```

And in `OutboundFileProcessorImpl.java` change:

```java
    @Value("${darts.audio.audio_gap_seconds:1}")
    private int acceptableAudioGapSecs;
```

to

```java
    @Value("${darts.audio.audio_gap_duration}")
    private Duration allowableAudioGapDuration;
```

And blow that change through all code usages/interfaces, down to your method here, which would simplify to:

```java
    private boolean hasGapBetweenAudios(AudioFileInfo audioFileInfoFirst, 
                                        AudioFileInfo audioFileInfoNext, 
                                        Duration allowableAudioGap) {
        if (audioFileInfoFirst == null || audioFileInfoNext == null) {
            return true;
        }

        Duration actualGap = Duration.between(audioFileInfoFirst.getEndTime(), audioFileInfoNext.getStartTime());
        return actualGap.compareTo(allowableAudioGap) > 0;
    }
```

Reply:
switched to using Duration as per recommendation. Added extra unit test